### PR TITLE
[bitnami/airflow] Consistency on Airflow configuration & init improvements

### PR DESCRIPTION
**Pull request by juan131**
### Description of the change

This PR changes how Airflow configuration is adapted based on users parameters: instead of setting a bunch of env. vars and relying on main containers' init logic to prepare the config file(s), everything (except config properties containing credentials) is prepared by the chart in the ConfigMap, then init containers are used to complete configuration with config properties containing credentials.

The goals is to ensure the `airflow.cfg` file is the same on every Airflow component (scheduler, worker, web server, etc.) given currently there are small discrepancies. Also, similar to https://github.com/bitnami/charts/pull/30246, applying changes related to credentials on init-containers has important benefits on bootstrapping, issues isolation and resource usage efficiency.

### Benefits

- Configuration consistency
- Improved bootstrapping

### Possible drawbacks

None

### Applicable issues

- fixes https://github.com/bitnami/containers/issues/74359

### Additional information

Other changes included:

- "wait-for-web-server` init container is replaced by "wait-for-db-migrations" which is more accurate.
- `pod_template.yaml` is moved to its own ConfigMap.
- Simplifications on helper libraries.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/airflow/templates/web/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  name: {{ template "airflow.web.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" (dict "value" .Values.commonAnnotations "context" $) | nindent 4 }}
+  {{- end }}
+data:
+  webserver_config.py: |-
+  {{- if .Values.web.configuration }}
+    {{- include "common.tplvalues.render" (dict "value" .Values.web.configuration "context" .) }}
```
Comment (fmulero) [inline bitnami/airflow/templates/web/configmap.yaml:19]:
```suggestion
    {{- include "common.tplvalues.render" (dict "value" .Values.web.configuration "context" .) | indent 4 }}
```

**Review by fmulero — COMMENTED** [review]
Great @juan131 

I've just left a suggestion. Could you give it a look?

**Comment by nordz0r** [comment]
Not fixed for docker-compose delivery
image

**Comment by carrodher** [comment]
> Not fixed for [docker-compose](https://github.com/bitnami/containers/blob/main/bitnami/airflow/docker-compose-ldap.yml) delivery image

Please be aware that docker-compose.yaml has not undergone internal testing. Consequently, we advise its use exclusively for development or testing purposes. For production-ready deployments, we highly recommend utilizing its associated [Bitnami Helm chart](https://github.com/bitnami/charts/tree/main/bitnami/airflow).

If you detect any issue in the docker-compose.yaml file, feel free to contribute with a fix by following our [Contributing Guidelines](https://github.com/bitnami/containers/blob/main/CONTRIBUTING.md).

**Pull request merged**